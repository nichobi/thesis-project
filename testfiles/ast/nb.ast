Expression ::= Term;

abstract Term;
True   : Term;
False  : Term;
If     : Term ::= Cond:Term Then:Term Else:Term;
Zero   : Term;
Succ   : Term ::= Term;
Pred   : Term ::= Term;
IsZero : Term ::= Term;

