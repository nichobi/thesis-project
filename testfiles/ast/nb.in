[T-True]
True : Bool

[T-False]
False : Bool

[T-If]
t1: Bool,
t2: t,
t3: t
------------------
If(t1, t2, t3): t

[T-Zero]
Zero : Int

[T-Succ]
t1: Int
------------------
Succ(t1): Int

[T-Pred]
t1: Int
------------------
Pred(t1): Int

[T-IsZero]
t1: Int
------------------
IsZero(t1): Bool

