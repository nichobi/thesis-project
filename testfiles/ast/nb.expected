RuleSet
  List
    Rule Name="T-True"
      HasType
        Value ID="True"
        TyVal ID="Bool"
      List
    Rule Name="T-False"
      HasType
        Value ID="False"
        TyVal ID="Bool"
      List
    Rule Name="T-If"
      HasType
        Function ID="If"
          List
            Value ID="t1"
            Value ID="t2"
            Value ID="t3"
        TyVar ID="t"
      List
        HasType
          Value ID="t1"
          TyVal ID="Bool"
        HasType
          Value ID="t2"
          TyVar ID="t"
        HasType
          Value ID="t3"
          TyVar ID="t"
    Rule Name="T-Zero"
      HasType
        Value ID="Zero"
        TyVal ID="Int"
      List
    Rule Name="T-Succ"
      HasType
        Function ID="Succ"
          List
            Value ID="t1"
        TyVal ID="Int"
      List
        HasType
          Value ID="t1"
          TyVal ID="Int"
    Rule Name="T-Pred"
      HasType
        Function ID="Pred"
          List
            Value ID="t1"
        TyVal ID="Int"
      List
        HasType
          Value ID="t1"
          TyVal ID="Int"
    Rule Name="T-IsZero"
      HasType
        Function ID="IsZero"
          List
            Value ID="t1"
        TyVal ID="Bool"
      List
        HasType
          Value ID="t1"
          TyVal ID="Int"
